### DISCUSSION TITLE
Limit the automatic connection to the RustDesk serve

### DISCUSSION THREAD

**Discussion by JuneLee2024** [rustdesk/rustdesk#9744] [General]
Hello,
After installing the customer service, if the id server is not set, it will automatically connect to the Rustdesk server.
I want to limit the client to only connect to my server, not the RustDesk server.
However, limiting port 21116 to only access my server IP will show that it cannot connect. This is only possible when the IP restrictions on TCP and UDP traffic on port 21116 are released.
Will this bring security issues? 
How should I limit the automatic connection to the RustDesk server?
How can I make this restriction on the server?
Please do not provide a way to change the client, I do not have the ability to change it.
Thank you.

**Comment by xlionjuan** [comment]
If your client is set to connect your own server, it won't connect public server.

**Comment by JuneLee2024** [comment]
Is there no other way?
Because some customers forget to fill in our server and connect to Rustdesk's server. They either complain about the poor connection quality or worry about security issues.
Thanks.

**Reply by xlionjuan** [reply]
Prepare the detailed documentation and sent it to end users, this is what you ***should*** do.


For Windows users, you can try: https://rustdesk.com/docs/en/self-host/client-configuration/#5-put-config-in-rustdeskexe-file-name-windows-only

**Comment by Poordeveloper** [comment]
Custom client in Pro can help you, but I guess you do not want to pay 1 cent.

**Comment by JuneLee2024** [comment]
Yes.
You can tell me directly that the open source version will not provide this feature. Unless you pay, this security issue will never be solved. This is our design.

**Comment by JuneLee2024** [comment]
There is a security flaw in this design.
Even if a private server is pre-configured, in the event that, for any reason, the private server becomes unreachable, the client will attempt to connect to the public server.
Some may argue that this is not an issue.
However, for others, it is indeed a significant concern.
From a design perspective, this is a security vulnerability.
